### PR TITLE
fix(bootstrap_list): add port parsing

### PR DESCRIPTION
**Pull request by michael-zapata**
Enable the use of multiple `bucketd`s on a single host

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -1,9 +1,5 @@
 {
   "bootstrap": [
-    "connectora.ringr2.devsca.com",
-    "connectorb.ringr2.devsca.com",
-    "connectorc.ringr2.devsca.com",
-    "connectord.ringr2.devsca.com",
-    "connectore.ringr2.devsca.com"
+      "bucketclient.testing.local:9000"
```
Comment (DavidPineauScality) [inline config.json]: YES ! Finally rid of our connector's FQDNs from the test configs.. Also, this means we never used our dummy server, right ? :/

File: lib/RESTClient.js [hunk 2]
```diff
@@ -23,6 +24,23 @@ function errorMap(mdError) {
...
+ *
+ * @param {String[]} list - the list to process
+ *
+ * @return {{ host: string, port: number }[]}
+ */
+function parseBootstrapList(list) {
+    return list.map(elem => {
+        const tmp = elem.split(':');
+        const host = tmp[0];
+        const port = Number.parseInt(tmp[1], 10);
```
Comment (DavidPineauScality) [inline lib/RESTClient.js]: So we're assuming that the port is **_always_**  specified? Should we not make this optional, and default to the default port (9000) ?

File: config.json (hunk 1 above)
Comment (michael-zapata) [inline config.json]: I'm not really sure, the tests are started using the `testing.local` stuff. I still wonder why tests broke partly without that change.

File: lib/RESTClient.js (hunk 2 above)
Comment (michael-zapata) [inline lib/RESTClient.js]: Having a default would simplify greatly my work by not having to deal with side projects at first, but I think we should be strict when everything is up-to-date. Implicit configuration is the kind of gotcha we do not want to deal with later on.

Comment (DavidPineauScality) [inline lib/RESTClient.js]: fair enough. OK for me :)
Thanks for the answer, federation should generate it properly anyways for our private deployments.

**Comment by DavidPineauScality** [comment]
@ironman-machine validate that federation breaks with this new change ?
SCALITY_FEDERATION_BRANCH=ft/dep-patcher-ansible

**Comment by ThibaultRiviere** [comment]
@mvaude  The Pr is that ok for you test ? can we merge it ?

**Comment by ThibaultRiviere** [comment]
@ironman-machine  check that 
SCALITY_FEDERATION_BRANCH=ft/dep-patcher-ansible
SCALITY_INTEGRATION_BRANCH=ft/failure-tests

**Comment by RachedBenMustapha** [comment]
The changes look fine per se, but as @ThibaultRiviere asks, what about #80 ? It's been explicitly requested that the failover code be removed from bucketclient, which makes the bootstrap list concept kind of moot. The port parsing part is still useful though.

**Comment by michael-zapata** [comment]
That was orthogonal to this PR, this only adds parsing of the port while ensuring that old formats still work.

**Comment by ThibaultRiviere** [comment]
@michael-zapata  @RachedBenMustapha @DavidPineauScality  can we merge that ?

**Comment by DavidPineauScality** [comment]
This needs to be merged synchronously with some changes of (at least) Federation (and maybe S3 ?)
I'm leaving that up to @michael-zapata and @mvaude to tell me whether the CI actually validated this (I remember @mvaude creating branches to attempt using it with the dependency patcher among other things).

**Comment by michael-zapata** [comment]
@DavidPineauScality I ensured that no breaking change was done in any repo, providing defaults if need be. We can add these later on.

@ironman-machine Let's see if this works as intended.
SCALITY_FEDERATION_BRANCH=ft/dep-patcher-ansible
SCALITY_INTEGRATION_BRANCH=ft/failure-tests

**Comment by ThibaultRiviere** [comment]
Any updates on this one?

**Comment by ThibaultRiviere** [comment]
We should also change the branch i guess ```rel/1.1``` should not be supported anymore

**Comment by michael-zapata** [comment]
@ironman-machine Thoughts?

**Comment by ThibaultRiviere** [comment]
@ironman-machine

**Comment by ThibaultRiviere** [comment]
@michael-zapata  is that ready?

**Comment by michael-zapata** [comment]
@ThibaultRiviere End-to-end passed.

**Pull request merged**